src/job/http.go

```diff
@@ -149,7 +151,8 @@ func fastHTTPJob(ctx context.Context, args config.Args, globalConfig *GlobalConf
 		fasthttp.ReleaseResponse(resp)
 	}()
 
-	resp.SkipBody = true
+	// TODO: DiscardBody here
+	resp.SkipBody = false
```

Comment:
If I'm not mistaken this is only responsible for not reading response body back, can you provide more details about how it actually breaks the protocol?
And in any case, it's better to make it configurable via the job parameters, not sure which section it would go to thought

Reply:
Actually, if there's a need to not read responses there already a way to do it via packetgen so we can probably just remove this parameter and return to a version where we don't allocate response at all and pass nil to `client.Do` instead. It's not ideal as fasthttp allocates it behind the scenes anyway but at least it'll reduce the amount of code we have to support

Reply:
> If I'm not mistaken this is only responsible for not reading response body back, can you provide more details about how it actually breaks the protocol?

Lol it's obvious that the sever responds with body, so you have to read it...

> it's better to make it configurable via the job parameters

It would be better to just `conn.Close()` instead of having broken connection for the next request with fasthttp.

Reply:
unless http server implements tcp socket itself with the ability to tap into system level acknowledgements there's no way for it to determine whether the client has received the response or not and it can only write responses to it until the network layer buffer is overflown. If you don't care about validity of the code you might generate more load by not waiting for the response, especially if you are doing an L4 attack (the benefit of generating plausible traffic for an L4 attack is questionable but still).

I agree that it's better to have a way to send more correct traffic as it has more chance of reaching the server and I removed this setting (we can still do those L4 attacks with packetgen)

Reply:
> you might generate more load by not waiting for the response

Yeah, that's fun too :wink: :stuck_out_tongue_winking_eye: